modules/core/src/test/scala/org/tessellation/infrastructure/trust/storage/TrustStorageSuite.scala

```diff
@@ -0,0 +1,388 @@
+import eu.timepit.refined.numeric.NonNegative
+import org.scalacheck.Gen
+import weaver.SimpleIOSuite
+import weaver.scalacheck.Checkers
+
+object TrustStorageSuite extends SimpleIOSuite with Checkers {
+  test("initial trust storage is empty") {
+    for {
+      store <- TrustStorage.make
+      trust <- store.getTrust
+    } yield expect.all(isEmpty(trust))
+  }
+
+  test("internal trust storage is updated with trust labels") {
+    val gen = Gen.listOfN(4, genPeerObservationAdjustmentUpdate).map(labels => PeerObservationAdjustmentUpdateBatch(labels))
+
+    forall(gen) { labels =>
+      for {
+        store <- TrustStorage.make
+        firstTrust <- store.getTrust
+
+        _ <- store.updateTrust(labels)
+
+        expected = labels.updates.map(update => update.id -> TrustInfo(trustLabel = update.trust.value.some)).toMap
+        actual <- store.getTrust.map(_.trust)
+      } yield
+        expect.all(
+          isEmpty(firstTrust),
+          expected.toList === actual.toList
+        )
+    }
+  }
+
+  test("internal trust storage is updated with predicted trusts") {
+    val gen = Gen.mapOfN(4, genPeerLabel)
+
+    forall(gen) { trusts =>
+      for {
+        store <- TrustStorage.make
+        firstTrust <- store.getTrust
+
+        _ <- store.updatePredictedTrust(trusts)
+
+        expected = trusts.view.mapValues(trustValue => TrustInfo(predictedTrust = trustValue.some)).toMap
+        actual <- store.getTrust.map(_.trust)
+      } yield
+        expect.all(
+          isEmpty(firstTrust),
+          expected === actual
+        )
+    }
+  }
+
+  test("internal trust storage is updated with public trust") {
+    val gen = for {
+      peerId <- peerIdGen
+      labels <- Gen.mapOfN(4, genPeerLabel)
+    } yield (peerId, PublicTrust(labels))
+
+    forall(gen) {
+      case (peerId, publicTrust) =>
+        for {
+          store <- TrustStorage.make
+          firstTrust <- store.getTrust
+
+          _ <- store.updatePeerPublicTrustInfo(peerId, publicTrust)
+
+          expected = PublicTrustMap(Map(peerId -> PublicTrust(publicTrust.labels)))
+          actual <- store.getTrust.map(_.peerLabels)
+        } yield
+          expect.all(
+            isEmpty(firstTrust),
+            expected === actual
+          )
+    }
+  }
+
+  test("initial current ordinal trust storage is empty") {
+    for {
+      store <- TrustStorage.make
+      trust <- store.getCurrentOrdinalTrust
+    } yield expect.all(OrdinalTrustMap.empty === trust)
+  }
+
+  test("initial next ordinal trust storage is empty except for the ordinal") {
+    for {
+      store <- TrustStorage.make
+      trust <- store.getNextOrdinalTrust
+    } yield
+      expect.all(
+        isEmpty(trust.trust),
+        SnapshotOrdinal(ordinalTrustUpdateInterval) === trust.ordinal,
+        PublicTrustMap.empty === trust.peerLabels
+      )
+  }
+
+  test("next ordinal trust storage doesn't update from internal storage because expected ordinal has not been reached") {
```

Comment:
`because` -> `when expected ordinal has not been reached` or `before expected ordinal has been reached`

Reply:
Changed to `next ordinal trust storage doesn't update from internal storage before expected ordinal has been reached`